tests/failpoints/cases/test_disk_full.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use core::panic;
+use raftstore::store::msg::*;
+use std::time::Duration;
+use test_raftstore::*;
+use tikv_util::config::*;
+
+/*
+Disk full test will have 3 scenarios.
+[leader full]
+1. not allowed: business wirte, region merge/split,etc.
+2. allowed: read, config change(add/remove peer), transfer leader etc.
+
+[minority full]
+1. not allowed: log entry append
+2. allowed: config change,transfer leader,etc.
+
+[majority full]
+1. equal to minority full, no additional design.
+
+*/
+const DISK_FULL_LEADER: &str = "disk_full_leader";
```

Comment:
Better to rename `leader`, `follower_1` to `peer_1`, `peer_2`, ...

Reply:
done